### PR TITLE
Static GH integration

### PR DESCRIPTION
**Pull request by SHaaD94**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2896)

### CODE REVIEW

File: src/main/scala/beam/router/graphhopper/BeamPathDetailsBuilderFactory.scala [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package beam.router.graphhopper
+
+import java.util
+
+import com.graphhopper.routing.ev.EncodedValueLookup
+import com.graphhopper.routing.weighting.Weighting
+import com.graphhopper.util.details.{PathDetailsBuilder, PathDetailsBuilderFactory}
+
+class BeamPathDetailsBuilderFactory extends PathDetailsBuilderFactory {
```
Comment (SHaaD94) [inline src/main/scala/beam/router/graphhopper/BeamPathDetailsBuilderFactory.scala:9]: Remove

File: src/main/scala/beam/router/graphhopper/GraphHopperWrapper.scala [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                          id2Link: Map[Int, (Coord, Coord)]
+                        ) extends Router {
+  private val graphHopper = {
+    val profiles = GraphHopperWrapper.getProfiles(carRouter)
+    val graphHopper = new BeamGraphHopper(wayId2TravelTime)
+    graphHopper.setGraphHopperLocation(graphDir)
+    graphHopper.setProfiles(profiles.asJava)
+    graphHopper.getCHPreparationHandler.setCHProfiles(profiles.map(p => new CHProfile(p.getName)).asJava)
+    graphHopper.importOrLoad()
+    //    val ghConfig = new GraphHopperConfig()
```
Comment (SHaaD94) [inline src/main/scala/beam/router/graphhopper/GraphHopperWrapper.scala:41]: remove comments

File: src/main/scala/beam/router/graphhopper/OriginalIdDetails.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package beam.router.graphhopper
+
+import com.graphhopper.routing.querygraph.VirtualEdgeIteratorState
+import com.graphhopper.util.{EdgeIteratorState, GHUtility}
+import com.graphhopper.util.details.AbstractPathDetailsBuilder
+
+class OriginalIdDetails extends AbstractPathDetailsBuilder("original_edge_id") {
```
Comment (SHaaD94) [inline src/main/scala/beam/router/graphhopper/OriginalIdDetails.scala:7]: remove

File: src/main/scala/beam/router/r5/RoutingWorker.scala [hunk 4]
```diff
@@ -127,11 +150,39 @@ class R5RoutingWorker(workerParams: WorkerParameters) extends Actor with ActorLo
       askForMoreWork()
 
     case request: RoutingRequest =>
-      msgs += 1
+      msgs.incrementAndGet()
       if (firstMsgTime.isEmpty) firstMsgTime = Some(ZonedDateTime.now(ZoneOffset.UTC))
       val eventualResponse = Future {
         latency("request-router-time", Metrics.RegularLevel) {
-          r5.calcRoute(request)
+          if (!request.withTransit && (carRouter == "staticGH" || carRouter == "quasiDynamicGH")) {
```
Comment (SHaaD94) [inline src/main/scala/beam/router/r5/RoutingWorker.scala:157]: remove withTransit

File: src/main/scala/beam/router/r5/RoutingWorker.scala [hunk 5]
```diff
@@ -142,6 +193,16 @@ class R5RoutingWorker(workerParams: WorkerParameters) extends Actor with ActorLo
       askForMoreWork()
 
     case UpdateTravelTimeLocal(newTravelTime) =>
+      log.info("===================================================================")
+      log.info(s"TOTAL NON TRANSIT ROUTING REQUESTS: ${routeRequestCounter.get()}, TOTAL EXECUTION TIME ${routeRequestExecutionTime.get()}, TOTAL REQUESTS ${msgs.get()}")
```
Comment (SHaaD94) [inline src/main/scala/beam/router/r5/RoutingWorker.scala:197]: remove debug logging?

File: test/input/beamville/beam.conf [hunk 6]
```diff
@@ -8,7 +8,7 @@ include "../common/matsim.conf"
 beam.agentsim.simulationName = "beamville"
 beam.agentsim.agentSampleSizeAsFractionOfPopulation = 1.0
 beam.agentsim.firstIteration = 0
-beam.agentsim.lastIteration = 2
+beam.agentsim.lastIteration = 5
```
Comment (SHaaD94) [inline test/input/beamville/beam.conf:11]: switch back

File: test/input/sf-light/sf-light-1k.conf [hunk 7]
```diff
@@ -3,6 +3,9 @@ include "../common/akka-router.conf"
 include "../common/metrics.conf"
 include "../common/matsim.conf"
 
+//beam.routing.carRouter="quasiDynamicGH"
+beam.routing.carRouter="quasiDynamicGH"
```
Comment (SHaaD94) [inline test/input/sf-light/sf-light-1k.conf:7]: remove

File: test/input/sf-light/sf-light-1k.conf [hunk 8]
```diff
@@ -11,7 +14,7 @@ include "../common/matsim.conf"
 beam.agentsim.simulationName = "sf-light-1k-xml"
 beam.agentsim.agentSampleSizeAsFractionOfPopulation = 1.0
 beam.agentsim.firstIteration = 0
-beam.agentsim.lastIteration = 0
+beam.agentsim.lastIteration = 2
```
Comment (SHaaD94) [inline test/input/sf-light/sf-light-1k.conf:17]: switch back

**Comment by SHaaD94** [comment]
Test!

File: src/main/scala/beam/agentsim/agents/PersonAgent.scala [hunk 9]
```diff
@@ -1082,7 +1082,7 @@ class PersonAgent(
         log.warning(s"$id is still travelling at end of simulation.")
         log.warning(s"$id events leading up to this point:\n\t${getLog.mkString("\n\t")}")
       } else if (stateName == PerformingActivity) {
-        logger.warn(s"$id is performing Activity at end of simulation")
+        logger.debug(s"$id is performing Activity at end of simulation")
```
Comment (wrashid) [inline src/main/scala/beam/agentsim/agents/PersonAgent.scala:1085]: also keep logger.warn(s"Performing Activity at end of simulation")

File: src/main/scala/beam/router/r5/RoutingWorker.scala [hunk 10]
```diff
@@ -1,47 +1,60 @@
 package beam.router.r5
 
+import java.io.File
```
Comment (REASY) [inline src/main/scala/beam/router/r5/RoutingWorker.scala:3]: Shall this moved to `beam.router` namespace?

File: src/main/scala/beam/router/r5/RoutingWorker.scala [hunk 11]
```diff
@@ -53,7 +66,7 @@ class R5RoutingWorker(workerParams: WorkerParameters) extends Actor with ActorLo
 
   private val tickTask: Cancellable =
     context.system.scheduler.scheduleWithFixedDelay(2.seconds, 10.seconds, self, "tick")(context.dispatcher)
-  private var msgs: Long = 0
+  private val msgs = new AtomicLong()
```
Comment (REASY) [inline src/main/scala/beam/router/r5/RoutingWorker.scala:69]: Actor executes the received messages in single threaded mode (until you create Future or Thread manually). So it is not needed to have it as Atomic

File: src/main/scala/beam/router/r5/RoutingWorker.scala (hunk 5 above)
Comment (REASY) [inline src/main/scala/beam/router/r5/RoutingWorker.scala:197]: `msgs` will be set to zero, so seems like not very useful? (https://github.com/LBNL-UCB-STI/beam/pull/2896/files#diff-5467656850d0ed2888f4e04a54e89be6R130)

File: src/main/scala/beam/agentsim/agents/PersonAgent.scala (hunk 9 above)
Comment (SHaaD94) [reply src/main/scala/beam/agentsim/agents/PersonAgent.scala:1085]: done

File: src/main/scala/beam/router/r5/RoutingWorker.scala (hunk 5 above)
Comment (SHaaD94) [reply src/main/scala/beam/router/r5/RoutingWorker.scala:197]: fixed

Comment (SHaaD94) [reply src/main/scala/beam/router/r5/RoutingWorker.scala:197]: changed to debug

File: src/main/scala/beam/router/r5/RoutingWorker.scala (hunk 11 above)
Comment (SHaaD94) [reply src/main/scala/beam/router/r5/RoutingWorker.scala:72]: makes sense, for us it was previously used inside of future, so i've changed it to atomic
reverted it back now

File: src/main/scala/beam/router/r5/RoutingWorker.scala (hunk 10 above)
Comment (SHaaD94) [reply src/main/scala/beam/router/r5/RoutingWorker.scala:3]: done

**Comment by SHaaD94** [comment]
Test!

**Comment by SHaaD94** [comment]
Test!

**Comment by SHaaD94** [comment]
Test!

**Comment by SHaaD94** [comment]
Test!

**Comment by SHaaD94** [comment]
Test!

**Comment by SHaaD94** [comment]
Test!

**Comment by SHaaD94** [comment]
Test!

**Comment by SHaaD94** [comment]
Test!

**Comment by SHaaD94** [comment]
Test!

**Pull request merged**